job-engine/client/src/main/java/org/eclipse/kapua/job/engine/client/JobStartOptionsClient.java

```diff
@@ -56,6 +60,25 @@ public void removeTargetIdToSublist(KapuaId targetId) {
         getTargetIdSublist().remove(targetId);
     }
 
+    @Override
+    public List<JobStepProperty> getStepPropertiesOverrides() {
+        if (stepPropertiesOverrides == null) {
+            stepPropertiesOverrides = new ArrayList<>();
+        }
+
+        return stepPropertiesOverrides;
+    }
+
+    @Override
+    public void addStepPropertyOverride(JobStepProperty jobStepPropertyOverride) {
+        getStepPropertiesOverrides().add(jobStepPropertyOverride);
+    }
+
+    @Override
+    public void setStepPropertiesOverrides(List<JobStepProperty> stepPropertiesOverrides) {
+        this.stepPropertiesOverrides = stepPropertiesOverrides;
```

Comment:
consider copying the elements into the inner list, to avoid changing the list implementation with a different one, and side effects from the list changing externally